### PR TITLE
Remove function index printing

### PR DESCRIPTION
**Pull request by tlively**
`BinaryIndexes` was only used in two places (Print.cpp and
wasm-binary.h), so it didn't seem to be a great fit for
module-utils.h. This change moves it to wasm-binary.h and removes its
usage in Print.cpp. This means that function indexes are no longer
printed, but those were of limited utility and were the source of
annoying noise when updating tests, anyway.

### CODE REVIEW

File: test/binaryen.js/atomics.js.txt [hunk 1]
```diff
@@ -1,7 +1,8 @@
+atexit() called, but EXIT_RUNTIME is not set, so atexits() will not be called. set EXIT_RUNTIME to 1 (see the FAQ)
```
Comment (aheejin) [inline test/binaryen.js/atomics.js.txt:1]: I think these messages should be deleted? (this and other files) These are only printed when the binaryen is in debug mode.

Comment (tlively) [reply test/binaryen.js/atomics.js.txt:1]: Ah shoot, good catch.

**Comment by tlively** [comment]
When I discussed this with @kripken offline in the context of #2718, he indicated that he didn't personally find them useful. I don't either, but if you or someone else feels that they are useful, I'd be happy to leave them in. This change is just a cleanup, so I don't think it's worth losing useful functionality.

**Comment by kripken** [comment]
I don't find them that useful for debugging myself. But if you or others use them, we should reconsider this patch.

I like the patch because it's very rare to need those indexes in my debugging experience at least. And wabt's tools can provide the indexes for those cases.

**Comment by aheejin** [comment]
I think this should be fine too.

**Pull request merged**